src/anemoi/training/train/forecaster.py

```diff
@@ -127,6 +128,23 @@ def __init__(
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         return self.model(x, self.model_comm_group)
 
+    @staticmethod
+    def get_loss_function(config: DictConfig, **kwargs) -> torch.nn.Module:
+        """
+        Get loss function from config.
+
+        Will include additional kwargs if specified in the config.
+
+        E.g.
+            If `include_node_weights: True` is set in the config, and node_weights in kwargs
+             `node_weights` will be included in the config to instantiate the module with.
+        """
+        config = dict(config)
+        for key in kwargs:
+            if config.pop(f"include_{key}", False):
```

Comment:
I don't think we should start the pattern of modifying config entries, this will get unwieldy very quickly.

Reply:
What do you propose for dynamic setting of the initialisation config with calculated arguments?

Reply:
Similar pattern to what we use for the Encoder / Decoder / Processor, use config items as config items and code inputs as code inputs.

Reply:
Are you able to review the updated code? It may address your concerns, but if not I am happy to reevaluate.